.github/scripts/pr-review.cjs

```diff
@@ -0,0 +1,114 @@
+/**
+ * This script will automatically check all PRs in the repository, and for any that have
+ *  the 'needs-review' label *and* a review from a contributor posted after
+ * the date of the most recent commit, it will remove the 'needs-review' label.
+ *
+ * This can be tested locally with the Github CLI installed, with:
+ * set GH_TOKEN=**** GH_REPO=OverlayPlugin/cactbot
+ * node ./.github/scripts/pr-review.cjs
+ */
+'use strict';
+
+const github = require('@actions/github');
+const { execSync } = require('child_process');
+
+/**
+ * @typedef {ReturnType<typeof import("@actions/github").getOctokit>} GitHub
+ */
+
+/**
+ * @param {GitHub} github
+ * @param {string} owner
+ * @param {string} repo
+ * @returns {Promise<void>}
+ */
+const checkAllPRs = async (github, owner, repo) => {
+  // Start by grabbing all PRs (including closed), as the workflow that fires this script
+  // will have latency and otherwise wouldn't pick up PRs that are approved and merged
+  // before the script has time to complete.
+  const iterator = github.paginate.iterator(github.rest.pulls.list, {
+    owner: owner,
+    repo: repo,
+    per_page: 100, // eslint-disable-line camelcase
+    state: 'all',
+  });
+
+  // iterate through each response
+  console.log('Fetching PRs...');
+  for await (const { data: prs } of iterator) {
+    for (const pr of prs) {
+      const prNumber = pr.number;
+
+      // check if the PR has a `needs-review` label; if not, skip for efficiency
+      const prLabels = pr.labels;
+      const hasNeedsReviewLabel = prLabels
+        .map((label) => label.name)
+        .includes('needs-review');
+      if (!hasNeedsReviewLabel)
+        continue;
```

Comment:
Can we pre-filter for this label on the initial request instead? I'm not super familiar with the interfaces here, but a quick google suggests it might be possible?

Reply:
> Can we pre-filter for this label on the initial request instead? I'm not super familiar with the interfaces here, but a quick google suggests it might be possible?

I remember looking at this initially, and just did a little follow-up digging to confirm - but I don't believe the `Oktokit.js` interface to the GH REST API supports a `label` parameter, at least according to [the docs](https://octokit.github.io/rest.js/v20#pulls).  Do you happen to have a link for what you were looking at?

I'm reasonably sure you can pass a label value as a param if, e.g., you do a curl request and include it in the URL string, but, ehhh, I guess I'm not crazy about abandoning the Oktokit lib and figuring out how to handle the env vars, doing an http get, and then parsing the json?  Unless you had something different in mind?

Reply:
At the start of the docs:

> Custom request options can be passed as request.* options. See [@octokit/request](https://www.npmjs.com/package/@octokit/request#octokitrequest) options. The same options can be passed to each endpoint request method.

I believe that this implies that adding a `labels` parameter to the request should work (also based on other documentation for `@octokit/request` and the actual GH APIs)? e.g.:
```js
  const iterator = github.paginate.iterator(github.rest.pulls.list, {
    owner: owner,
    repo: repo,
    per_page: 100, // eslint-disable-line camelcase
    state: 'all',
    labels: 'needs-review',
  });
```

Reply:
> At the start of the docs:
> 
> > Custom request options can be passed as request.* options. See [@octokit/request](https://www.npmjs.com/package/@octokit/request#octokitrequest) options. The same options can be passed to each endpoint request method.
> 
> I believe that this implies that adding a `labels` parameter to the request should work (also based on other documentation for `@octokit/request` and the actual GH APIs)? e.g.:
> 
> ```js
>   const iterator = github.paginate.iterator(github.rest.pulls.list, {
>     owner: owner,
>     repo: repo,
>     per_page: 100, // eslint-disable-line camelcase
>     state: 'all',
>     labels: 'needs-review',
>   });
> ```

Nice catch in the docs - didn't see that buried sentence.  But unfortunately, it doesn't seem to accept that param at least with pulls.list - tried with both `labels:` and `request: { labels: }` in my test script, and it's still yanking down all of the PRs.  But I'll keep poking at it.

Reply:
> Nice catch in the docs - didn't see that buried sentence. But unfortunately, it doesn't seem to accept that param at least with pulls.list - tried with both `labels:` and `request: { labels: }` in my test script, and it's still yanking down all of the PRs. But I'll keep poking at it.

I've narrowed down at least part of the issue: the `pulls` endpoint doesn't support filtering by label (even though `issues` apparently does).  You can actually validate this right now at https://api.github.com/repos/OverlayPlugin/cactbot/pulls?labels=needs-review -- even with this param, it's still including PR 90, even though it doesn't have that label.

I've looked at a couple of the other methods, including `issuesAndPullRequests` and the general `search` method, but -- and I can't tell if this is just because of the way `actions/github` implemented the oktokit API or whether it's actually an oktokit limitation, they either don't play nicely with the pagination iterator or they don't return the full PR data (including reviews and commits) that the `pulls` endpoint does.  Open to figuring out a way to do this, but I'm running low on ideas...

Reply:
> Can we pre-filter for this label on the initial request instead? I'm not super familiar with the interfaces here, but a quick google suggests it might be possible?

@valarnin Figured out a way to do this using a different method; tested and working now.